app/Database/Migrations.hs

```diff
@@ -0,0 +1,37 @@
+module Database.Migrations
+    (migrateDatabase) where
+
+import Control.Monad.Reader (MonadIO)
+import Database.Database (getDatabaseVersion)
+import Database.Persist.Sql (Migration, SqlPersistT, addMigration, runMigration, runMigrationUnsafe)
+import Database.Tables (migrateAll)
+
+data MigrationWrapper = MigrationWrapper {
+    version :: Int,
+    script :: Migration
+}
+
+-- | Migrates the database
+migrateDatabase :: MonadIO m => SqlPersistT m ()
+migrateDatabase = do
+    -- Run unsafe migrations
+    currVersion <- getDatabaseVersion
+    applyMigrations currVersion migrationList
+    -- Run safe migrations
+    runMigration migrateAll
```

Comment:
I'm not sure that `migrateAll` makes sense here, though maybe I'm missing something. `Database.Tables` will contain the "definitive" version of the current schema, but it should be equivalent to applying the written migrations one by one (from the base version of the schema currently on master).

In other words, if the migrations are written correctly, this step should be unnecessary; but if the migrations aren't written correctly, there's a bigger issue that shouldn't be fixed by executing this `migrateAll`.